include/seastar/core/memory.hh

```diff
@@ -443,15 +443,20 @@ struct allocation_site {
 /// @return a vector of \ref allocation_site
 std::vector<allocation_site> sampled_memory_profile();
 
-/// @brief Copies the current sampled set of allocation_sites in the output parameter
-/// (up to size of output vector)
-///
-/// Returns amount of copied elements. Useful if one wants to avoid allocating
-/// an output vector (e.g.: under OOM conditions)
-///
-/// @param output vector to copy the allocation sites to
+/// @brief Copies the current sampled set of allocation_sites into the
+/// array pointed to by the output parameter
+/// 
+/// Copies up to \p size elements of the current sampled set of allocation
+/// sites into the output array, which must have length at least \p size.
+///
+/// Returns amount of copied elements. This method does not allocation to
+/// it is useful alternative to \ref sampled_memory_profile() when one wants
+/// to avoid allocating (e.g.: under OOM conditions).
+///
+/// @param output array to copy the allocation sites to
+/// @param size the size of the array pointed to by \p output
 /// @return number of \ref allocation_site copied to the vector
-size_t sampled_memory_profile(std::vector<allocation_site>& output);
+size_t sampled_memory_profile(allocation_site* output, size_t size);
```

Comment:
Please fold this into the original patch.

Reply:
This is now folded in force push: https://github.com/scylladb/seastar/compare/58e4eb2dd3d67da55c0b76a186f045cd47ac3996..7a97ddfda2fec056975f00f20a8859c25bac41d9